subsys/bluetooth/services/ots/ots_oacp.c

```diff
@@ -298,102 +298,93 @@ static enum bt_gatt_ots_oacp_res_code oacp_proc_validate(
 	}
 };
 
-static enum bt_gatt_ots_oacp_res_code oacp_command_decode(
-	const uint8_t *buf, uint16_t len,
-	struct bt_gatt_ots_oacp_proc *proc)
+static int oacp_command_decode(const uint8_t *buf, uint16_t len,
+			       struct bt_gatt_ots_oacp_proc *proc)
 {
 	struct net_buf_simple net_buf;
 
 	net_buf_simple_init_with_data(&net_buf, (void *) buf, len);
 
+	__ASSERT(len >= OACP_PROC_TYPE_SIZE,
```

Comment:
Where is the length coming from? Does it depend on data coming over the air? We should never assert in such cases but handle the error gracefully, whereas asserts should be used to catch issues that indicate local bugs in the implementation.

Reply:
The length is indeed coming over the air, but the calling function `bt_gatt_ots_oacp_write` has a check to ensure we never try to decode the message if the length is not at least large enough to have the opcode present. This assert was added as part of a conversation @Thalley and I had [here](https://github.com/zephyrproject-rtos/zephyr/pull/41470#discussion_r780127907) just to prevent any future bugs.

But clearly, this is not a good solution since the question came up again. I have removed the assert, and moved the checks inside of the decode functions. I wanted to provide the maximum amount of logging so that is why there are different error codes for when the opcode is decoded vs when there is not enough data to even decode the opcode.